latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/TemplateProcessor.java

```diff
@@ -159,4 +165,38 @@ public String render(String templateName) throws Exception
     {
         return render(getTemplate(templateName));
     }
+
+    private String getTemplateName(Block block)
+    {
+        // If there's a custom template defined in the Block parameter's, use it!
+        String templateName = block.getParameter("latex-template");
+        if (templateName == null) {
+            templateName = block.getClass().getSimpleName();
+        }
+        return templateName;
+    }
+
+    private Optional<String> renderUIXPs(String templateName, String suffix)
+    {
+        // The XDOM template case is particular, the after UIXP needs to be located before "\end{document}" which closes
+        // the latex document. To do so, the UIXP of XDOM are directly integrated in the template in Velocity.
+        Optional<String> ret;
```

Comment:
Small detail: I like to call it "result" (if that's what you meant by "ret"). At least we agreed in the past to not use abbreviations (since they're misleading and can be understood differently depending on the reader. Ret = return ? Ret = retrieve ? Ret = retaliation ? :)).